src/datasources/balances-api/coingecko-api.service.spec.ts

```diff
@@ -1191,4 +1192,266 @@ describe('CoingeckoAPI', () => {
       `Error getting native coin price: Error: pricesProvider.nativeCoinId is not defined `,
     );
   });
+
+  describe('batching price requests', () => {
+    function buildCoinGeckoResponse(
+      tokenAddresses: Array<string>,
+      fiatCode: string,
+    ): AssetPrice {
+      const lowerCaseFiatCode = fiatCode.toLowerCase();
+      return tokenAddresses.reduce<AssetPrice>((acc, tokenAddress) => {
+        const price = faker.number.float({ min: 0.01, multipleOf: 0.01 });
+        const change = faker.number.float({ min: -1, max: 1 });
+        acc[tokenAddress] = {
+          [lowerCaseFiatCode]: price,
+          [`${lowerCaseFiatCode}_24h_change`]: change,
+        };
+        return acc;
+      }, {});
+    }
+    it('should request prices from network', async () => {
+      const chainName = faker.company.name();
+      const tokenAddresses = faker.helpers.multiple(
+        () => faker.finance.ethereumAddress(),
+        {
+          count: {
+            min: 1,
+            max: CoingeckoApi.MAX_BATCH_SIZE,
+          },
+        },
+      );
+      const fiatCode = faker.finance.currencyCode();
+      const lowerCaseFiatCode = fiatCode.toLowerCase();
+      const coingeckoPrice = buildCoinGeckoResponse(tokenAddresses, fiatCode);
+      mockNetworkService.get.mockResolvedValue({
+        data: rawify(coingeckoPrice),
+        status: 200,
+      });
+
+      const result = await service._requestPricesFromNetwork({
+        chainName,
+        tokenAddresses,
+        fiatCode,
+      });
+
+      expect(result).toStrictEqual(coingeckoPrice);
+      expect(mockNetworkService.get).toHaveBeenCalledTimes(1);
+      expect(mockNetworkService.get).toHaveBeenNthCalledWith(1, {
+        url: `${coingeckoBaseUri}/simple/token_price/${chainName}`,
+        networkRequest: {
+          headers: {
+            'x-cg-pro-api-key': coingeckoApiKey,
+          },
+          params: {
+            contract_addresses: tokenAddresses.join(','),
+            vs_currencies: lowerCaseFiatCode,
+            include_24hr_change: true,
+          },
+        },
+      });
+    });
+
+    it('should deduplicate token addresses', async () => {
+      const chainName = faker.company.name();
+      const tokenAddress = faker.finance.ethereumAddress();
```

Comment:
Nit: To make the test case cover more scenarios we can test it with a couple of tokenAddresses instead of one.
```suggestion
      const tokenAddressesToDeduplicate = [faker.finance.ethereumAddress(), faker.finance.ethereumAddress()];
```

Reply:
This test ensures the same address isn't used twice. By using two different addresses, there is inherently no deduplication.

Reply:
I meant that we should check for the deduplication of multiple items in an array, rather than just a single item. I'm thinking about scenarios where the method might remove the first duplicate but miss subsequent ones. Anyway, it's more of a nit, so I'll leave it up to your discretion.

Reply:
Changed in ec0e96b8067c84c356bc2c1f74270db7bfa6147c.